lib/league_stats.rb

```diff
@@ -0,0 +1,124 @@
+class LeagueStats
+  attr_accessor :games, :teams
+
+  def initialize(games, teams)
+    @games = games
+    @teams = teams
+  end
+
+  def sort_total_league_games
+    @teams.each do |team|
+      @games.each do |game|
+        if game.home_team_id == team.team_id
+          home_team_league_stat_collector(team, game)
+        elsif game.away_team_id == team.team_id
+          away_team_league_stat_collector(team, game)
+        end
+      end
+    team_league_totals(team)
+    end
+  end
+
+  def home_team_league_stat_collector(team, game)
+    team.home_goals_league += game.home_goals
+    team.total_games_league += 1
+    team.home_games_league += 1
+    team.total_goals_allowed_league += game.away_goals
+    if game.home_goals > game.away_goals
+      team.home_wins_league += 1
+    end
+  end
+
+  def away_team_league_stat_collector(team, game)
+    team.away_goals_league += game.away_goals
+    team.total_games_league += 1
+    team.away_games_league += 1
+    team.total_goals_allowed_league += game.home_goals
+    if game.away_goals > game.home_goals
+      team.away_wins_league += 1
+    end
+  end
+
+  def team_league_totals(team)
+    team.total_wins_league = team.home_wins_league + team.away_wins_league
+    team.total_games_league = team.home_games_league + team.away_games_league
+    team.total_goals_league = team.home_goals_league + team.away_goals_league
+  end
+
+  def highest_offense
+    @teams.max_by do |team|
+      if team.calculate_offense.class != Float
```

Comment:
@nagerz can you explain what this line is doing? Won't team.calculate_offense always return a float?

Reply:
That was my attempt at fool proofing the code to not need a complete set of games data (i.e., make it work when dividing by zero.) It didn't work, so I built out the game test data set. I left that in as a reminding to look at it with a teacher or someone.

Reply:
Gotcha. I think an alternative might be
`if team.calculate_offense == 0`
Ruby will read 0.0 as equal to 0 but still as a Float which is why it wasn't working